src/NHibernate.Test/NHSpecificTest/DtcFailures/DtcFailuresFixture.cs

```diff
@@ -113,8 +118,8 @@ public void Can_roll_back_transaction()
 		[Description("Another action inside the transaction do the rollBack outside nh-session-scope.")]
 		public void RollbackOutsideNh()
```

Comment:
This one showcase the roadblock: session close within an ongoing transaction scope is rejected by connection manager.

The connection manager close does close any pending explicit transaction, then call disconnect which checks there are no pending transaction, explicit or ambient. The ambient check on disconnect should be removed, ambient transaction supports connection disconnect.

This need changes in NHibernate project even for testing:  the connection manager is not pluggable.